TTBot/Commands/ChampionshipsModule.cs

```diff
@@ -0,0 +1,212 @@
+            {
+                var author = Context.Message.Author as SocketGuildUser;
+                if (!await _permissionService.UserIsModeratorAsync(Context, author))
+                {
+                    await Context.Channel.SendMessageAsync("You dont have permission to create events");
+                    return;
+                }
+
+                // clear out our results DB table first
+                await _results.DeleteAllAsync<ChampionshipResultsModel>();
+
+                var attachment = Context.Message.Attachments.First();
+                var excelData = await _excelService.ReadResultsDataFromAttachment(attachment);
+
+                List<ChampionshipResultsModel> championshipResults = new List<ChampionshipResultsModel>();
+
+                foreach (ExcelDataModel excelDataModel in excelData)
+                {
+                    var e = await _events.GetEventByShortname(excelDataModel.Championship);
+
+                    if (e == null || e.Id == 0)
+                    {
+                        if (!listOfUnknownChampionships.Contains(excelDataModel.Championship))
+                        {
+                            listOfUnknownChampionships.Add(excelDataModel.Championship);
+                        }
+                        continue;
+                    } else
+                    {
+                        if (!listOfSuccessfulyUploadedChampionships.Contains(excelDataModel.Championship))
+                        {
+                            listOfSuccessfulyUploadedChampionships.Add(excelDataModel.Championship);
+                        }
+                    }
+
+                    var eventId = e.Id;
+                    var points = 0;
+
+                    foreach (String position in excelDataModel.Positions)
+                    {
+                        int p;
+                        if (position.Contains("*"))
+                        {
+                            points += ChampionshipPointsModel.FastestLapPoints;
+                            p = Int32.Parse(position.Remove(position.IndexOf("*"), 1));
+                        }
+                        else
+                        {
+                            p = Int32.Parse(position);
+                        }
+
+                        points += ChampionshipPointsModel.PointsAwarded[p];
+                    }
+
+                    ChampionshipResultsModel championshipResult = new ChampionshipResultsModel()
+                    {
+                        EventId = eventId,
+                        Driver = excelDataModel.Driver,
+                        Positions = excelDataModel.Positions,
+                        Points = points
+                    };
+
+                    championshipResults.Add(championshipResult);
+
+                }
+
+                _results.AddAsync(championshipResults);
+
+
+            } catch (Exception ex)
+            {
+                sb.AppendLine($"Error when doing import: {ex.Message}");
```

Comment:
maybe it'd be better to have this try catch inside the loop so that it can try to parse the remaining sheets if one fails